core/src/main/java/org/apache/iceberg/stats/BlobMetadata.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+
+public class BlobMetadata {
+  private final String type;
+  private final List<Integer> columns;
+  private final long fileOffset;
+  private final long length;
+  private final Optional<String> compressionCodec;
+
+  @JsonCreator
+  public BlobMetadata(
+      @JsonProperty("type") String type,
+      @JsonProperty("columns") List<Integer> columns,
+      @JsonProperty("offset") long fileOffset,
+      @JsonProperty("length") long length,
+      @JsonProperty("compression_codec") Optional<String> compressionCodec) {
+    this.type = Objects.requireNonNull(type, "type is null");
+    this.columns = ImmutableList.copyOf(Objects.requireNonNull(columns, "columns is null"));
+    this.fileOffset = fileOffset;
+    this.length = length;
+    this.compressionCodec = Objects.requireNonNull(compressionCodec, "compressionCodec is null");
+  }
+
+  public String getType() {
```

Comment:
Iceberg avoids the use of `get` in method names. In cases like this one, it's an unnecessary additional word. And in most other cases, there's a more specific verb that should be used instead, like `find` or `fetch` that give the reader more context about what's happening.